All_home_works/hw10/task1_get_result.py

```diff
@@ -0,0 +1,140 @@
+"""
+Ваша задача спарсить информацию о компаниях, находящихся в индексе S&P 500 с данного сайта:
+https://markets.businessinsider.com/index/components/s&p_500
+
+Для каждой компании собрать следующую информацию:
+
+    Текущая стоимость в рублях (конвертацию производить по текущему курсу, взятому с сайта центробанка РФ)
+    Код компании (справа от названия компании на странице компании)
+    P/E компании (информация находится справа от графика на странице компании)
+    Годовой рост/падение компании в процентах (основная таблица)
+    Высчитать какую прибыль принесли бы акции компании (в процентах),
+     если бы они были куплены на уровне 52 Week Low
+      и проданы на уровне 52 Week High (справа от графика на странице компании)
+
+Сохранить итоговую информацию в 4 JSON файла:
+
+    Топ 10 компаний с самими дорогими акциями в рублях.
+    Топ 10 компаний с самым низким показателем P/E.
+    Топ 10 компаний, которые показали самый высокий рост за последний год
+    Топ 10 комппаний, которые принесли бы наибольшую прибыль,
+     если бы были куплены на самом минимуме и проданы на самом максимуме за последний год.
+    Пример формата:
+
+    [
+    {
+     "code": "MMM",
+     "name": "3M CO.",
+     "price" | "P/E" | "growth" | "potential profit" : value,
+    },
+    ...
+    ]
+
+
+    P.S. по-максимуму использовать возможности параллелизма и асинхронности.
+
+    bs4
+    aiohttp
+
+"""
+import asyncio
+from multiprocessing import Pool
+
+import aiohttp
+from bs4 import BeautifulSoup
+
+URL = "https://markets.businessinsider.com/index/components/s&p_500"
+MAIN_URL = "https://markets.businessinsider.com/"
+COURSE_URL = "https://www.cbr.ru/scripts/XML_daily.asp"
+COURSE = 1
+
+
+async def fetch_response(url):
+    async with aiohttp.ClientSession() as session:
+        async with session.get(url) as response:
+            return await response.text()
+
+
+async def collect_html_for_companies(data):
+    company_url, annual_growth = data
+    company_url = MAIN_URL + company_url
+    return await fetch_response(company_url), annual_growth
+
+
+async def get_current_course():
+    def _parse_current_course(html):
+        soup = BeautifulSoup(html, features="lxml")
+        course = soup.find("valute", id="R01235").value.get_text(
+            strip=True, separator=""
+        )
+        return float(course.replace(",", "."))
+
+    response = await fetch_response(COURSE_URL)
+    return _parse_current_course(response)
+
+
+async def collect_data():
+    global COURSE
+    response, COURSE = await asyncio.gather(fetch_response(URL), get_current_course())
```

Comment:
`response` name describes the technical side of variable: something that comes from the internet.   

I prefer to see that name that describes its purpose.    You get responses from currency, from the company list, and from the company page.   Since they all in the same project, it's better to give them unique and descriptive names.